### PR TITLE
Header a11y fix

### PR DESCRIPTION
**Pull request by xman343**
## Changes

https://github.com/iTwin/iTwinUI/issues/1148 - The only example `Main` is currently failing. This is due to the lack of an accessible label for the `HeaderSplitButton`. 
* Passing solution: [Implemented ](https://github.com/iTwin/iTwinUI/pull/1495/commits/8fd0c97e498ffa3443c973a7a2bfe0755ca4d689) `aria-label` "More" for `HeaderSplitButton`. This helps the screen reader communicate the purpose of the button to the user.
## Testing

Testing will be conducted through the Cypress web a11y script. Any breaking changes will be evaluated through unit testing & visual tests. Example also read through NVDA to assess screen reader compatibility. 

## Docs

TBD

### CODE REVIEW

**Comment by xman343** [comment]
Conflicted on whether to implement the accessible name through `VisuallyHidden` or a `title` for the button. I'm going with `title` for now since it serves the extra function of displaying the label in a popover when it's hovered on. I could change it if we decide that it's unnecessary for this component.

File: packages/itwinui-react/src/core/Header/HeaderSplitButton.tsx [hunk 1]
```diff
@@ -52,6 +52,7 @@ export const HeaderSplitButton = React.forwardRef((props, forwardedRef) => {
         onHide={React.useCallback(() => setIsMenuOpen(false), [])}
       >
         <ButtonBase
+          title='Dropdown menu button'
```
Comment (mayank99) [inline packages/itwinui-react/src/core/Header/HeaderSplitButton.tsx:55]: please don't use `title`. https://www.tpgi.com/using-the-html-title-attribute-updated/

`aria-label` should work fine.

and "Dropdown menu button" is not good as the label; but at the same time we don't really know what will be inside the dropdown, so the user should be the one providing it.

i'd say lets go with just "More" for now. in the future, we can make it customizable (#1444)

```suggestion
          aria-label='More'
```

File: .changeset/ninety-geckos-roll.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@itwin/itwinui-react': patch
+---
+
+Added accessible tille to dropdown menu button in HeaderSplitButton.
```
Comment (mayank99) [inline .changeset/ninety-geckos-roll.md:5]:
```suggestion
Added accessible name to dropdown menu button in HeaderSplitButton.
```

File: packages/itwinui-react/src/core/Header/HeaderSplitButton.tsx (hunk 1 above)
Comment (xman343) [reply packages/itwinui-react/src/core/Header/HeaderSplitButton.tsx]: Thanks for sharing the link. I didn't realize that `title` was that problematic.

**Review by mayank99 — APPROVED** [review]

**Review by siddhantrawal — APPROVED** [review]

**Pull request merged**